src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py

```diff
@@ -126,10 +126,20 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
             resource = resources.get_by_id(
                 cluster_resource_id, parent_api_version)
             cluster_location = resource.location.lower()
-            if resource.properties['totalNodeCount'] == 1 or resource.properties['totalNodeCount'] == 2:
-                configuration_settings['clusterPurpose'] = 'DevTest'
-            if resource.properties['distribution'].lower() == "openshift":
-                configuration_settings[self.OPEN_SHIFT] = "true"
+            try:
+                if cluster_type.lower() == 'connectedclusters':
+                    if resource.properties['totalNodeCount'] < 3:
+                        configuration_settings['clusterPurpose'] = 'DevTest'
+                    if resource.properties['distribution'].lower() == "openshift":
```

Comment:
This is passed from the extension manager into the chart directly. Why do you need this in the `configuration_settings`?

Reply:
Fixed